From ba192f5f8cdca310dfe28b02cfd7cf615fc1d1a0 Mon Sep 17 00:00:00 2001
From: arathinx <ayyappanx.rathinam@intel.com>
Date: Thu, 11 Jul 2019 11:42:57 +0800
Subject: [PATCH] DRVLIB_SW-2536: Assign the DMA controller ID instead of
 controller enum value in CQM

Currently DMA controller enum value is filled in the param dma_chan of struct cbm_dp_alloc_data() while doing the dp_port_alloc(). Now we have to fill the DMA controller ID instead of DMA controller enum value
---
 drivers/net/ethernet/lantiq/cqm/prx300/cqm.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/lantiq/cqm/prx300/cqm.c b/drivers/net/ethernet/lantiq/cqm/prx300/cqm.c
index b94e39a5fc2b..f42c2aeac29a 100644
--- a/drivers/net/ethernet/lantiq/cqm/prx300/cqm.c
+++ b/drivers/net/ethernet/lantiq/cqm/prx300/cqm.c
@@ -1916,7 +1916,8 @@ static void fill_dp_alloc_data(struct cbm_dp_alloc_data *data, int dp,
 			p_info->dma_ch = chan;
 			snprintf(p_info->dma_chan_str, DMA_CH_STR_LEN, "port%d",
 				 (port + idx));
-			data->dma_chan = p_info->dma_ch;
+			data->dma_chan = _DMA_C(p_info->dma_dt_ctrl, 0,
+						p_info->dma_dt_ch);
 			data->tx_pkt_credit = p_info->dq_txpush_num;
 		} else {
 			data->flags |= CBM_PORT_PKT_CRDT_SET;
